src/core/tsi/openssl_utils.cc

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <absl/strings/string_view.h>
+#include "openssl/bio.h"
+#include "openssl/evp.h"
+#include "openssl/pem.h"
+#include "openssl/x509.h"
+
+#include "src/core/tsi/openssl_utils.h"
+
+namespace grpc_core {
+
+OpenSslPKey::OpenSslPKey(absl::string_view private_key) {
+  BIO* bio = BIO_new_mem_buf(private_key.data(), private_key.size());
+  p_key_ = PEM_read_bio_PrivateKey(bio, nullptr, nullptr, nullptr);
+  BIO_free(bio);
+}
+
+OpenSslX509::OpenSslX509(absl::string_view cert) {
```

Comment:
nit: Would it make sense to use cert_chain here instead of cert?

Reply:
I used `cert` cos `OpenSslX509` only contains a single certificate. Even \when passed a `cert_chain`, it just uses the first cert.

Reply:
Since that is an internal implementation, I suggest using cert_chain as the parameter name here.

Reply:
SG. It will be in the next push.